### PR TITLE
React Hooks Support

### PR DESCRIPTION
**Pull request by JordanW7**
### Purpose
- Reduce technical debt and legacy code
- Improve application performance
- Simplify implementation, `presentView` is no longer required
- Support a full end to end React Hooks based application (which has endless benefits and opens up various possibilities, such as being able to use hooks at the root level and easily manage entire application state with hooks/context, rather than having to make 'hack' like solutions around the previous implementation).

#### view.js
```js
import React from 'react';
import { render } from 'react-dom';
import { registerSiftView } from '@redsift/sift-sdk-web';
import { useSiftView } from '@redsift/sift-sdk-web/react';

const View = () => {
  const [params, siftView] = useSiftView();
  if (!params) {
    return null;
  }
  const [{ rpcApiConfig, routeLocation }] = params.data;
  return (
    <App
      siftView={siftView}
      rpcApiConfig={rpcApiConfig}
      routeLocation={routeLocation}
    />
  );
};

export default View;

const ViewRender = render(<View />, document.getElementById('root'));
registerSiftView(ViewRender);
```

### Other:
- Added `prettier` + `eslint` precommit configuration
- Deleted `yarn.lock` - shouldn't have two different kinds of lock files for any repo, and this file hadn't been updated for 2+ years (unlike `package-lock.json`).
- Replaced `@redsift/rollup-bundler` with the latest version of `rollup` - `@redsift/rollup-bundler` is significantly outdated, completely unnecessary, and doesn't support multiple file input/output.
- changed the npm publishing to only publish the build files, rather than every single file in the repository.

### Test
- Implemented in https://github.com/redsift/sift-organisation-manager/pull/129 (basic sift)
- Implemented in https://github.com/redsift/sift-oninbox/pull/592 (complex sift, including a public sift)
- Backwards compatibility checked here: https://github.com/redsift/sift-oninbox/commit/c1d9033726544c28fb6624593c5042d9f15b2514

### CODE REVIEW

**Comment by JordanW7** [comment]
`useSiftView` complete & tested.

**Comment by JordanW7** [comment]
No idea why CircleCI is failing 
```
npm ERR! Cannot read property 'match' of undefined
 ```

File: src/index.js [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
 
 export { EmailClientController };
 export { SiftController };
 export { SiftStorage };
 export { SiftView };
+export { useSiftController };
+export { useSiftView };
 
 /**
  * SiftView
```
Comment (JordanW7) [inline src/index.js:16]: I don't understand the purpose of `registerSiftView` and `registerSiftController`, they just log to console... but I've left them in the usage docs for the new functions just in case...

**Comment by JordanW7** [comment]
This should be good to go now.

**Comment by JordanW7** [comment]
- Implemented in redsift/sift-organisation-manager#129 (basic sift)
- Implemented in redsift/sift-oninbox#592 (complex sift, including a public sift)
- Backwards compatibility checked here: redsift/sift-oninbox@c1d9033

Should be good to go now.

File: package.json [hunk 2]
```diff
@@ -27,7 +29,32 @@
...
+    "husky": ">=4",
+    "lint-staged": ">=10",
+    "prettier": "^2.0.5",
+    "rollup": "^2.10.9"
+  },
+  "peerDependencies": {
+    "react": "^16.13.1",
+    "react-dom": "^16.13.1"
+  },
+  "husky": {
```
Comment (karlnorling) [inline package.json:51]: _scope creep_: Should we add `engine` declaring build support for 12 and up?
```
"engines": {
    "node": ">=12.14.0",
    "npm": ">=6.13.4"
  },
```
See:

https://github.com/redsift/sift-oninbox-admin/blob/master/frontend/package.json#L12-L15

**Comment by deepakprabhakara** [comment]
I think registerSiftView and registerSiftController are no-ops, but we'll need to test. Can do it in the next release. Nice work @JordanW7, only small change is 12.16.3 as the node version instead of 12.14.0. @karlnorling Once this is pushed through and all ok we should change the sift templates in the SDK as well.
